includes/class-omise-capabilities.php

```diff
@@ -68,6 +72,27 @@ public static function retrieve($pKey = null, $sKey = null)
 		return self::$instance;
 	}
 
+	/**
+	 * @return boolean
+	 */
+	public static function shouldCallApi() {
+		$omiseSettingPages = [ 'omise' ];
+
+		$currentAdminPage = isset( $_GET[ 'page' ] ) ? $_GET[ 'page' ] : '';
+
+		// If page is checkout page but not thank you page.
+		// By default thank you page is also part of checkout pages
+		// and we do not need to call capabilities on thank you page.
+		// If endpoint url is `order-received`, it mean thank you page.
+		$isPaymentPage = is_checkout() && !is_wc_endpoint_url( 'order-received' );
+
+		// If page is omise setting page from admin panel.
+		$isOmiseSettingPage = is_admin() && in_array( $currentAdminPage, $omiseSettingPages );
```

Comment:
Small, but just for the sake of readability, I would move this line to just after `$omiseSettingPages` and `$currentAdminPage` are defined.